lib/index.rb

```diff
@@ -77,8 +81,8 @@ def first_error?(error)
     billing_phone_number: billing_phone_number,
     billing_email_address: billing_email_address
   }
-  save_customer_details = SaveCustomerDetails.new(customer_gateway: @customer_gateway)
-  response = save_customer_details.execute(customer_details: @customer_details)
+  response = @duplicate_address.execute(customer_details: @customer_details) if use_same_address
+  response = @save_customer_details.execute(customer_details: @customer_details) unless use_same_address
```

Comment:
Should there be two response variables?

Reply:
They both have conditionals at the end, so there will only be one assigned

Reply:
Oh I get now, miss understood the conditional.

Reply:
Should lines 84 and 85 be shortened to 
`  response =  use_same_address ? @duplicate_address.execute(customer_details: @customer_details) : @save_customer_details.execute(customer_details: @customer_details)
`

Reply:
I thought that would make the line too long. But if you think its better I can change it?